wgpu-core/src/device/life.rs

```diff
@@ -814,6 +835,63 @@ impl<A: HalApi> LifetimeTracker<A> {
                 }
             }
         }
+
+        if !self.suspected_resources.blas_s.is_empty() {
+            let (mut guard, _) = hub.blas_s.write(token);
+            let mut trackers = trackers.lock();
+
+            for id in self.suspected_resources.blas_s.drain(..) {
+                if trackers.blas_s.remove_abandoned(id) {
+                    log::debug!("Blas {:?} will be destroyed", id);
+                    #[cfg(feature = "trace")]
+                    if let Some(t) = trace {
+                        t.lock().add(trace::Action::DestroyBlas(id.0));
+                    }
+
+                    if let Some(res) = hub.blas_s.unregister_locked(id.0, &mut *guard) {
+                        let submit_index = res.life_guard.life_count();
+                        self.active
+                            .iter_mut()
+                            .find(|a| a.index == submit_index)
+                            .map_or(&mut self.free_resources, |a| &mut a.last_resources)
+                            .acceleration_structures
+                            .extend(res.raw);
+                    }
+                }
+            }
+        }
+
+        if !self.suspected_resources.tlas_s.is_empty() {
+            let (mut guard, _) = hub.tlas_s.write(token);
+            let mut trackers = trackers.lock();
+
+            for id in self.suspected_resources.tlas_s.drain(..) {
+                if trackers.tlas_s.remove_abandoned(id) {
```

Comment:
This line appears to be causing a memory leak, it seems to always return false, looking in this function, it seems 
````rust 
existing_ref_count.load()
````
in 
````rust
if existing_epoch == epoch && existing_ref_count.load() == 1 {
````
 always returns 2 not 1 and so this is returning false (moving code out of the if stament at this line seems to fix the memory leak)